Code/Tools/ProjectManager/Source/DownloadController.cpp

```diff
@@ -99,10 +102,13 @@ namespace O3DE::ProjectManager
             succeeded = false;
         }
 
-        DownloadableObject downloadableObject = m_objects.front();
-        m_objects.erase(m_objects.begin());
-        emit Done(downloadableObject.m_objectName, succeeded);
-        emit ObjectDownloadRemoved(downloadableObject.m_objectName, downloadableObject.m_objectType);
+        if (!m_objects.empty())
+        {
+            DownloadableObject downloadableObject = m_objects.front();
```

Comment:
nitpick: You can move the DownloadableObject from `m_objects.front()` to save on a copy operation.